appl/include/obj_parser.h

```diff
@@ -1,181 +1,570 @@
-            obj->vt[vt_index].x = atof(token);
+    obj->triangles = NULL;
+    obj->triangle_amount = 0;
+    obj->triangles_capacity = STARTING_DYNAMIC_ARRAY_CAPACITY;
+    obj->vertex_amount = 0;
+    obj->position_amount = 0;
+    obj->uv_amount = 0;
+    obj->normal_amount = 0;
 
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vt[vt_index].y = atof(token);
+    return obj;
+}
 
-            vt_index++;
-        }
+void __obj_free(obj_t* obj)
+{
+    free(obj->triangles);
 
-        if (strncmp(buffer, "vn ", 3) == 0)
+    free(obj);
+}
+
+int __obj_add_triangle(obj_t* obj, obj_triangle_t* triangle)
+{
+    if(obj->triangle_amount != 0)
+    {
+        if(obj->triangle_amount >= obj->triangles_capacity)
         {
-            char *remaining_tokens;
-            char *token;
+            obj->triangles_capacity *= STARTING_DYNAMIC_ARRAY_CAPACITY;
+            obj->triangles = (obj_triangle_t*)realloc(obj->triangles, (obj->triangles_capacity) * sizeof(obj_triangle_t));
+        }
+    }
+    else
+    {
+        obj->triangles = (obj_triangle_t*)calloc(STARTING_DYNAMIC_ARRAY_CAPACITY, sizeof(obj_triangle_t));
+    }
 
-            strtok_s(buffer, " ", &remaining_tokens); // vt token
+    if(!obj->triangles)
+    {
+        return 1;
+    }
 
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vn[vn_index].x = atof(token);
+    obj->triangles[obj->triangle_amount] = *triangle;
+    obj->triangle_amount++;
 
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vn[vn_index].y = atof(token);
+    return 0;
+}
 
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vn[vn_index].z = atof(token);
+void __obj_update_other_counters(obj_t* obj, obj_info_t* obj_info)
+{
+    obj->vertex_amount = obj_info->vertex_amount;
+    obj->position_amount = obj_info->position_amount;
+    obj->uv_amount = obj_info->uv_amount;
+    obj->normal_amount = obj_info->normal_amount;
+}
 
-            vn_index++;
-        }
+obj_t* obj_parse(const char* file_name)
+{
+    FILE* file = __obj_file_open(file_name);
+    if(!file)
+    {
+        fprintf(stderr, "Error: Trying to open the file %s!\n", file_name);
+        return NULL;
+    }
 
-        if (strncmp(buffer, "f ", 2) == 0)
-        {
-            char *remaining_tokens;
-            char *token;
+    obj_info_t* obj_info = __obj_info_new();
+    if(!obj_info)
+    {
+        fprintf(stderr, "Error: Trying to allocate memory for obj_info!\n");
+        __obj_file_close(file);
+        return NULL;
+    }
 
-            strtok_s(buffer, " ", &remaining_tokens); // f token
+    obj_t* obj = __obj_new();
+    if(!obj)
+    {
+        fprintf(stderr, "Error: Trying to allocate memory for obj!\n");
+        __obj_file_close(file);
+        __obj_info_free(obj_info);
+        return NULL;
+    }
 
-            for (int i = 0; i < 3; i++)
+    char buffer[1024];
+    while(fgets(buffer, sizeof(buffer), file))
```

Comment:
the 4 `if` checking for a tag (v, vt, vn and f), are always executed (even if only one is validated to true)

Improve performance using 1 if and 3 else if.